esp/services/ws_dali/ws_daliservice.hpp

```diff
@@ -0,0 +1,78 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _ESPWIZ_ws_dali_HPP__
+#define _ESPWIZ_ws_dali_HPP__
+
+#include "ws_dali_esp.ipp"
+#include "exception_util.hpp"
+
+class CWSDaliEx : public CWSDali
+{
+    StringAttr espProcess;
+    IEspContainer* container = nullptr;
+    std::atomic<bool> daliDetached{false};
+
+public:
+    IMPLEMENT_IINTERFACE;
+
+    virtual void setContainer(IEspContainer* container)
+    {
+        container = container;
+    }
+
+    bool isDaliDetached()
+    {
+        return daliDetached;
+    }
+
+    bool attachServiceToDali() override
```

Comment:
minor: by convention, we usually declare virtual overrides with 'virtual' a well.

Reply:
are these common methods to many services, could they be commoned up in a base class?

Reply:
I will add the 'virtual'. 

We MAY add those common methods to a base class. Will add a separate JIRA.